src/CloudStructures/RedisConnection.cs

```diff
@@ -1,129 +1,157 @@
+            : this.GetConnection().GetDatabase());
+
+        /// <summary>
+        /// Gets a transaction.
+        /// </summary>
+        public ITransaction Transaction => ((IDatabase)Database).CreateTransaction();
+
+        /// <summary>
+        /// Gets target servers.
+        /// </summary>
+        public IServer[] Servers
+        {
+            get
+            {
+                if (this.Config != null)
+                {
+                    return this.Config.Options
+                        .EndPoints
+                        .Select(this.GetConnection(), static (x, c) => c.GetServer(x))
+                        .ToArray();
+                }
+                else
+                {
+                    var multiplexer = Database.Multiplexer;
+                    var endpoints = multiplexer.GetEndPoints();
 
+                    return endpoints.Select(endpoint => multiplexer.GetServer(endpoint))
+                        .ToArray();
+                }
+            }
+        }
+        #endregion
+
+
+        #region Constructors
+        /// <summary>
+        /// Creates instance with RedisConfig.
+        /// </summary>
+        /// <param name="config"></param>
+        /// <param name="converter">If null, use <see cref="SystemTextJsonConverter"/> as default.</param>
+        /// <param name="handler"></param>
+        /// <param name="logger"></param>
+        public RedisConnection(RedisConfig config, IValueConverter? converter = null, IConnectionEventHandler? handler = null, TextWriter? logger = null)
+        {
+            this.Config = config;
+            this.Converter = new(converter);
+            this.Handler = handler;
+            this.Logger = logger;
+        }
 
-    #region Constructors
-    /// <summary>
-    /// Creates instance.
-    /// </summary>
-    /// <param name="config"></param>
-    /// <param name="converter">If null, use <see cref="SystemTextJsonConverter"/> as default.</param>
-    /// <param name="handler"></param>
-    /// <param name="logger"></param>
-    public RedisConnection(RedisConfig config, IValueConverter? converter = null, IConnectionEventHandler? handler = null, TextWriter? logger = null)
-    {
-        this.Config = config;
-        this.Converter = new(converter);
-        this.Handler = handler;
-        this.Logger = logger;
-    }
-    #endregion
+        /// <summary>
+        /// Creates a RedisConnection instance with a pre-existing IDatabaseAsync instance.
+        /// </summary>
+        /// <param name="database">Pre-existing IDatabaseAsync instance.</param>
+        /// <param name="converter">Optional value converter.</param>
+        /// <param name="handler">Connection event handler.</param>
+        /// <param name="logger">Logger.</param>
+        public RedisConnection(IDatabaseAsync database, IValueConverter? converter = null, IConnectionEventHandler? handler = null, TextWriter? logger = null)
```

Comment:
ah okay so this is the main change in this PR then? Allowing the database to be passed in to the constructor. That makes sense.

In the code samples where you saw CloudStructures being used (both the customer one and the hello world one), I thought I saw some interactions with this class through static methods (meaning that the caller wouldn't necessarily have even called the constructor first). Is that accurate or am I misremembering?

Reply:
The hello world example, I have been calling the constructor to make a connection while in the customer code, they were directly using the "GetConnection" call from the RedisConnection class. But due to lack of full code, I couldn't determine if they were making a call to the same `GetConnection` as present in this RedisConnection class.